homeassistant/components/alpha_vantage/sensor.py

```diff
@@ -130,15 +130,18 @@ def update(self):
         _LOGGER.debug("Requesting new data for symbol %s", self._symbol)
         all_values, _ = self._timeseries.get_intraday(self._symbol)
         values = next(iter(all_values.values()))
-        self._attr_native_value = values["1. open"]
+        if isinstance(values, dict) and "1. open" in values:
+            self._attr_native_value = values["1. open"]
+        else:
+            self._attr_native_value = None
         self._attr_extra_state_attributes = (
             {
                 ATTR_ATTRIBUTION: ATTRIBUTION,
                 ATTR_CLOSE: values["4. close"],
                 ATTR_HIGH: values["2. high"],
                 ATTR_LOW: values["3. low"],
             }
-            if values is not None
+            if values is not None and isinstance(values, dict)
```

Comment:
A further improvement would be to only check once if `values` is valid. Now we do it potentially two times in the same method.

Reply:
this would result in one "if" inside the other which is less readable.

Reply:
I was thinking we can check once if values is valid and if not, return after setting native value and attributes to None.